src/components/common/FileUploadPdf.vue

```diff
@@ -0,0 +1,316 @@
+    // if (have file) and (required) => depends on file validation
+    // if (have file) and (not required) => depends on file validation
+
+    if (!file) {
+      if (this.isRequired) {
+        this.errorMessages = ['File is required']
+        return
+      } else {
+        this.errorMessages = []
+        return
+      }
+    }
+
+    // validate the file
+    this.errorMessages = ['Processing...']
+    const validFile = await this.validateFile(file)
+    if (!validFile) return
+
+    // upload the file
+    this.errorMessages = ['Uploading...']
+    const fileKey = await this.uploadFile(file)
+    if (!fileKey) return
+
+    // update parent
+    this.updateFile(file)
+    this.updateFileKey(fileKey)
+
+    // if we get this far then everything succeeded
+    this.errorMessages = []
+  }
+
+  /**
+   * Validates the PDF file. Note that some of this is async.
+   * @param file the file to validate
+   * @returns whether file is valid
+   */
+  async validateFile (file: File): Promise<boolean> {
+    if (typeof file.arrayBuffer === 'undefined') return true
+
+    // verify file size
+    if (this.maxSize && (file?.size > (this.maxSize * 1024 * 1024))) {
+      this.errorMessages = [`Exceeds maximum ${this.maxSize} MB file size`]
+      return false
+    }
+
+    // try to retrieve file info
+    const fileInfo = await this.retrieveFileInfo(file).catch(() => null as PdfInfoIF)
+    if (fileInfo === null) {
+      this.errorMessages = ['Invalid PDF']
+      return false
+    }
+
+    // verify encryption
+    if (fileInfo.isEncrypted) {
+      this.errorMessages = ['File must be unencrypted']
+      return false
+    }
+
+    // verify content lock
+    if (fileInfo.isContentLocked) {
+      this.errorMessages = ['File content cannot be locked']
+      return false
+    }
+
+    // verify page sizes
+    if (this.pageSize) {
+      const valid = await this.isPageSize(file, this.pageSize).catch(() => null as boolean)
+      if (valid === null) {
+        this.errorMessages = ['Invalid PDF']
+        return false
+      }
+      if (!valid) {
+        this.errorMessages = [FileUploadPdf.PAGE_SIZE_DICT[this.pageSize].validationErrorMsg]
+        return false
+      }
+    }
+
+    // if we get this far then file is valid
+    return true
+  }
+
+  /**
+   * Retrieves encryption and content lock info from PDF file.
+   * Throws an exception on invalid PDF.
+   * @param file the file to check
+   * @return an object containing the file's info
+   */
+  async retrieveFileInfo (file: File): Promise<PdfInfoIF> {
+    try {
+      const pdfBufferData = await file.arrayBuffer()
+      const pdfData = new Uint8Array(pdfBufferData) // put it in a Uint8Array
+      const pdf = await this.pdfjsLib.getDocument({ data: pdfData }).promise
+      const perms = await pdf.getPermissions()
+      return { isEncrypted: false, isContentLocked: !!perms }
+    } catch (err) {
+      if (err.name === 'PasswordException') {
```

Comment:
In VS Code I get an error that "property name doesn't exist on type unknown). You can fix by casting err to any on this line or line 219,

Reply:
That works, too, but you're changing the original type of err.

Either way is OK with me.

Reply:
Looks good, thanks.